src/autoqasm/api.py

```diff
@@ -433,16 +441,26 @@ def _func(*args, **kwargs) -> Any:
     )
     _func.__annotations__[first_param.name] = first_param.annotation
 
+    lexer = OpenQASM3Lexer()
+
     new_params = [first_param]
     for param in sig.parameters.values():
         if param.annotation is param.empty:
             raise errors.MissingParameterTypeError(
                 f'Parameter "{param.name}" for subroutine "{_func.__name__}" '
                 "is missing a required type hint."
             )
+        # Check whether 'param.name' is a OpenQasm keyword
+        for tokenType, val in lexer.get_tokens(param.name):
+            if tokenType is Token.Keyword.Type and not hasattr(f, "__wrapped__"):
```

Comment:
this is awesome! minor note - I think it would be better if we defined a separate/global list of AutoQASM "reserved words", and checked that list here, rather than checking directly with the OQ3 lexer on the fly. (We may want to reserve additional words in AutoQASM.) It would be fine to hard-code that list, or obtain it from the OQ3 lexer for now - but ideally it should be a separate function call rather than inline.

Reply:
(because eventually we will want to do a similar check when creating classical variables in the program, in addition to the subroutine parameters that you've done here - and we'll want to easily reuse the logic there.)

Reply:
Ok sure

Reply:
Few of the keywords like `frame`, `cal`, `waveform`, `port` do not throw `QASM3ParsingError` after using them as variable name in the code from "steps to reproduce"

Reply:
Thanks for checking that! Even if the parser does not throw an error, we should still not allow those keywords as variable names - since by the spec they are technically not allowed.

Reply:
Ok